src/Tutor.Core/LearnerModel/DomainOverlay/KnowledgeComponentMasteries/AssessmentItemMastery.cs

```diff
@@ -58,5 +33,10 @@ private void When(AssessmentItemAnswered @event)
             SubmissionCount++;
             LastSubmissionTime = @event.TimeStamp;
         }
+
+        private void When(AssessmentItemEvent @event)
```

Comment:
Should the `AssessmentItemEvent` be abstract and not need a standalone When?

Reply:
It should be abstract, I forgot to mark it as such when I added the class. 

I think it makes sense to have a <code>When</code> method for it though, because it makes sense for KCM to apply any events that represent interactions with assessment items the same way - just forward them to the appropriate AIM, and if the mastery estimate for the assessment item changed, recalculate the mastery estimate for the knowledge component as well. The only way I see this changing in the future is if we introduce new mastery indicators which would also need to be updated. 

A more expressive name for <code>AssessmentItemEvent</code> might be useful though, since its meaning is something more like 'the learner interacted with the assessment item', but I'm not sure how to convey that without making it really long.